x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_esql.tsx

```diff
@@ -0,0 +1,769 @@
+  searchQueryLanguage: SEARCH_QUERY_LANGUAGE.KUERY,
+  filters: [],
+  showDistributions: true,
+  showAllFields: false,
+  showEmptyFields: false,
+  probability: null,
+  rndSamplerPref: 'off',
+  ...overrides,
+});
+
+export interface IndexDataVisualizerESQLProps {
+  getAdditionalLinks?: GetAdditionalLinks;
+}
+
+export const IndexDataVisualizerESQL: FC<IndexDataVisualizerESQLProps> = (dataVisualizerProps) => {
+  const { services } = useDataVisualizerKibana();
+  const { data, fieldFormats } = services;
+  const euiTheme = useCurrentEuiTheme();
+
+  const [query, setQuery] = useState<AggregateQuery>({ esql: '' });
+  const [currentDataView, setCurrentDataView] = useState<DataView | undefined>();
+
+  const updateDataView = (dv: DataView) => {
+    if (dv.id !== currentDataView?.id) {
+      setCurrentDataView(dv);
+    }
+  };
+  const [lastRefresh, setLastRefresh] = useState(0);
+
+  const _timeBuckets = useTimeBuckets();
+  const timefilter = useTimefilter({
+    timeRangeSelector: true,
+    autoRefreshSelector: true,
+  });
+
+  const indexPattern = useMemo(() => {
+    let indexPatternFromQuery = '';
+    if ('sql' in query) {
+      indexPatternFromQuery = getIndexPatternFromSQLQuery(query.sql);
+    }
+    if ('esql' in query) {
+      indexPatternFromQuery = getIndexPatternFromESQLQuery(query.esql);
+    }
+    // we should find a better way to work with ESQL queries which dont need a dataview
+    if (indexPatternFromQuery === '') {
+      return undefined;
+    }
+    return indexPatternFromQuery;
+  }, [query]);
+
+  const restorableDefaults = useMemo(
+    () => getDefaultDataVisualizerListState({}),
+    // We just need to load the saved preference when the page is first loaded
+
+    []
+  );
+
+  useEffect(
+    function updateAdhocDataViewFromQuery() {
+      let unmounted = false;
+
+      const update = async () => {
+        if (!indexPattern) return;
+        const dv = await getDataViewByIndexPattern(data.dataViews, indexPattern, currentDataView);
+
+        if (dv) {
+          updateDataView(dv);
+        }
+      };
+
+      if (!unmounted) {
+        update();
+      }
+
+      return () => {
+        unmounted = true;
+      };
+    },
+
+    [indexPattern, data.dataViews, currentDataView]
+  );
+
+  /** Search strategy **/
+  const fieldStatsRequest = useMemo(() => {
+    // Obtain the interval to use for date histogram aggregations
+    // (such as the document count chart). Aim for 75 bars.
+    const buckets = _timeBuckets;
+
+    const tf = timefilter;
+
+    if (!buckets || !tf || (isESQLQuery(query) && query.esql === '')) return;
+    const activeBounds = tf.getActiveBounds();
+
+    let earliest: number | undefined;
+    let latest: number | undefined;
+    if (activeBounds !== undefined && currentDataView?.timeFieldName !== undefined) {
+      earliest = activeBounds.min?.valueOf();
+      latest = activeBounds.max?.valueOf();
+    }
+
+    const bounds = tf.getActiveBounds();
+    const BAR_TARGET = 75;
```

Comment:
Is this consistent with the non-ES|QL view? Can they share the same const (assuming we don't want to use the `histogram:barTarget` Kibana advanced setting)?